src/zope/password/password.py

```diff
@@ -582,9 +571,105 @@ def match(self, hashed_password):
         :rtype: bool
         :returns: True iif the password was hashed with this manager.
         """
+        hashed_password = _encoder(hashed_password)
         return (hashed_password.startswith(self._prefix)
                 or self._z3c_bcrypt_syntax.match(hashed_password) is not None)
 
+class BCRYPTKDFPasswordManager(_PrefixedPasswordManager):
+    """
+    BCRYPT KDF password manager.
+
+    This manager converts a plain text password into a byte array .
+    The password and salt values (randomly generated when the password
+    is encoded) are combined and repeatedly hashed *rounds* times. The
+    repeated hashing is designed to thwart discovery of the key via
+    password guessing attacks. The higher the number of rounds, the
+    slower each attempt will be.
+
+    Compared to the :class:`BCRYPTPasswordManager`, this has the
+    advantage of allowing tunable rounds, so as computing devices get
+    more powerful making brute force attacks faster, the difficulty
+    level can be raised (for newly encoded passwords).
+
+    >>> from zope.password.password import BCRYPTKDFPasswordManager
+    >>> manager = BCRYPTKDFPasswordManager()
+    >>> manager.checkPassword(b'not from here', None)
+    False
+
+    Let's encode a password. We'll use the minimum acceptable number
+    of rounds so that the tests run fast:
+
+    >>> manager.rounds = 51
+    >>> password = u"right \N{CYRILLIC CAPITAL LETTER A}"
+    >>> encoded = manager.encodePassword(password)
+    >>> print(encoded.decode())
+    {BCRYPTKDF}33...
+
+    It checks out:
+
+    >>> manager.checkPassword(encoded, password)
+    True
+
+    We can change the number of rounds for future encodings:
+
+    >>> manager.rounds = 100
+    >>> encoded2 = manager.encodePassword(password)
+    >>> print(encoded2.decode())
+    {BCRYPTKDF}64...
+    >>> manager.checkPassword(encoded2, password)
+    True
+
+    And the old password still checks out:
+
+    >>> manager.checkPassword(encoded, password)
+    True
+    """
+
+    #: The number of rounds of hashing that should be applied.
+    #: The higher the number, the slower it is. It should be at least
+    #: 50.
+    rounds = 1<<10
```

Comment:
I'm wondering if `rounds = 2**10` wouldn't be a bit more Pythonic.

Reply:
Or even just the integer 1024. 

I just did what z3c.bcrypt did, although I reduced the rounds a bit. The original rounds was taking something like four seconds on my newish-computer, which seems excessive.